R/epiestim_rt.R

```diff
@@ -0,0 +1,191 @@
+##' @title Performs draws from the generation time distribution
+##'
+##' @param p A single [carehomes_parameters()] object.
+##'
+##' @param n The number of draws to perform
+##'
+##' @return a vector of `n` values of the generation time
+##'
+##' @importFrom distcrete distcrete
+##' @importFrom stats runif rpois
+gt_sample <- function(p, n = 1000) {
+
+  ## Note: the above does not account for the impact of
+  ## vaccination on the generation time.
+
+  if (length(unique(p$p_C)) > 1)  {
+    stop("gt_sample does not allow p_C to vary by age")
+  }
+
+  if (p$I_C_2_transmission > 0) {
+    stop("gt_sample does not allow transmission from I_C_2")
+  }
+  if (p$k_A > 1 || p$k_P  > 1 || p$k_C_1 > 1) {
+    stop("gt_sample does not allow k_A > 1, k_P > 1 or k_C_1 > 1")
+  }
+  if (p$I_P_transmission != 1 || p$I_C_1_transmission  != 1) {
+    stop("gt_sample does not allow
+    I_P_transmission !=1 or I_C_1_transmission != 1")
+  }
+
+  draw_from <- function(n, k, gamma) {
+    discrete_gamma <-
+      distcrete::distcrete("gamma", p$dt, shape = k, rate = gamma, w = 1)
+    discrete_gamma$r(n)
+  }
+
+  sampled_gt <- numeric(0)
+  i <- 0
+
+  while (i < n) {
+    ## Draw exposed period
+    sample_E <- draw_from(1, p$k_E, p$gamma_E)
+    ## Draw symptomatic vs asymptomatic path
+    path_C <- runif(1) <= p$p_C[[1]]
```

Comment:
```suggestion
    path_C <- runif(1) <= p$p_C
```
we already checked p$p_C is a scalar

Reply:
I think the check is that all the values in `p$p_C` are the same (`if (length(unique(p$p_C)) > 1`) but `p$p_C` is always a vector with 19 elements - so I think I should still use [[1]]

Reply:
@lwhittles  can you please check if you're happy with my response to this and the fix (from @richfitz) to your other comment about the sample issue? Once you're happy can you pls approve changes? thx!

Reply:
You're quite right!